gitops/snapshot.go

```diff
@@ -1058,16 +1058,30 @@ func FilterIntegrationTestScenariosWithContext(scenarios *[]v1beta2.IntegrationT
 	return &filteredScenarioList
 }
 
+func GetRequiredContextIntegrationTestScenario(scenarios []v1beta2.IntegrationTestScenario, contextName string) *[]v1beta2.IntegrationTestScenario {
```

Comment:
There is a potential issue with this function, because this will return only the integrationTestScenarios that have the `group` context, but won't return the ones which would also run for the group Snapshot, like the ones with the `application`/`all` context (which means the ITS runs in all cases, including group testing) as well as the `pull_request` context (which executes all ITS for an open Pull Request, which also includes group testing).

Reply:
I suppose [these code](https://github.com/konflux-ci/integration-service/pull/1049/files#diff-3540d1f95a378c2e6bea89135278f11b57b55c7c4e0a7ac892eb6dfd251b5b55R238-R251) can get other context ITS and also `group` ITS and join them together. But after thinking about again, it can't work well and also you remind me the checkRun's title for `group` snapshot is different from `component` snapshot. Let me fix it. Thanks!

Reply:
Updated by creating a temp group snapshot.